aws/aws/lib/lambdas/syncBlockProd.mjs

```diff
@@ -3,53 +3,74 @@
 const DynamoDB = require('aws-sdk/clients/dynamodb');
 const ApiGatewayManagementApi = require('aws-sdk/clients/apigatewaymanagementapi');
 
-// MAX_L2_BLOCKS : 20 -> 2 Tx per Blocka
-//               : 1 -> 32
-exports.handler = async (event) => {
-  const docClient = new DynamoDB.DocumentClient();
-  const sendEndpoint = process.env.SEND_ENDPOINT;
-  const MAX_L2_BLOCKS = 1;
+const docClient = new DynamoDB.DocumentClient();
+const sendEndpoint = process.env.SEND_ENDPOINT;
+const MAX_L2_BLOCKS = 1;
+const TYPE_TIMBER = 'timberProposed';
+const TYPE_BLOCK = 'blockProposed';
+
+exports.handler = async event => {
   const connectionId = event.requestContext.connectionId;
   const deployment = event.stageVariables.deployment;
 
   const eventBody = JSON.parse(event.body);
   console.log('Body', eventBody);
+
   const lastBlock = Number(eventBody.lastBlock);
   const type = eventBody.syncInfo || 'sync';
-  const api = new ApiGatewayManagementApi({ endpoint: `${sendEndpoint}${deployment.toLowerCase()}/`});
-  
+  const api = new ApiGatewayManagementApi({
+    endpoint: `${sendEndpoint}${deployment.toLowerCase()}/`,
+  });
 
-  const params3 = {
+  const paramsBlockLast = {
+    TableName: `PNF3_DocumentDB_${deployment}`,
+    Key: {
+      blockType: type === 'sync-timber' ? TYPE_TIMBER : TYPE_BLOCK,
+      blockNumberL2: lastBlock,
+    },
+  };
+  const paramsBlockNext = {
     TableName: `PNF3_DocumentDB_${deployment}`,
     Key: {
-      blockType: type === 'sync-timber'? 'timberProposed': 'blockProposed',
-      blockNumberL2: Number(lastBlock)+1,
+      blockType: type === 'sync-timber' ? TYPE_TIMBER : TYPE_BLOCK,
+      blockNumberL2: lastBlock + 1,
     },
   };
-  
-  const data = await docClient.get(params3).promise();
+
+  const [dataBlockLast, dataBlockNext] = await Promise.all([
```

Comment:
Reads tp DynamoDb Table are very expensive. Is it possible to read bock blocks with the same instruction?

Reply:
I just had a quick look at the Lambda metrics, and the duration of this Lambda execution didn't change at all. I think we should be fine. Exact queries like that are cheap and shouldn't affect the performance.